File: packages/tolk/test/parity/elementwise_3way/main.ml
(* Parity case: d = a + b + c, shape [256]. Tensor graph through rangeify. *)

open Tolk_uop
module U = Uop

let build () =
  let a = Helpers.mk_param ~idx:0 [ 256 ] in
  let b = Helpers.mk_param ~idx:1 [ 256 ] in
  let c = Helpers.mk_param ~idx:2 [ 256 ] in
  let ab = U.alu_binary ~op:Ops.Add ~lhs:a ~rhs:b in
  let abc = U.alu_binary ~op:Ops.Add ~lhs:ab ~rhs:c in
  Helpers.wrap_sink [ abc ]

let () =
  Helpers.dump_tensor
    ~stages:[ Helpers.Stage5; Helpers.Stage7 ]
    ~out_dir:Sys.argv.(1) (build ())

File: packages/tolk/test/parity/elementwise_3way/main.py
#!/usr/bin/env python3
"""Parity case: d = a + b + c, shape [256]."""

import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
from helpers import dump_tensor, mk_param, wrap_sink  # noqa: E402


def build():
    a = mk_param(0, 256)
    b = mk_param(1, 256)
    c = mk_param(2, 256)
    return wrap_sink(a + b + c)


if __name__ == "__main__":
    dump_tensor(build(), os.path.dirname(os.path.abspath(__file__)),
                stages=("stage5", "stage7"))
